features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/link/LinkChecker.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2025 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
+ * Please see LICENSE files in the repository root for full details.
+ */
+
+package io.element.android.features.messages.impl.link
+
+import com.squareup.anvil.annotations.ContributesBinding
+import io.element.android.libraries.core.data.tryOrNull
+import io.element.android.libraries.core.extensions.containsRtLOverride
+import io.element.android.libraries.di.AppScope
+import io.element.android.wysiwyg.link.Link
+import java.net.URI
+import javax.inject.Inject
+
+interface LinkChecker {
```

Comment:
I'm not sure there is a need here for an interface, we probably always want to test using the real implementation

Reply:
interface everywhere! it helps to unit test the presenter. But OK in this case, this is probably a bit too much.